### ISSUE TITLE
Editing a cell while grouped columns are enabled results in blank lines being inserted

### ISSUE THREAD

**Issue by scottmac991** [plotly/dash-ag-grid#215]
After editable fields are modified within a grid setup with row groupings, extra blank lines (and blank group) is added. These multiply by the number of times YOU try to edit. 

I've recreated the issue by slightly modifying the example in AG Grid Enterprise -> Group Cell Renderer, by making the City field editable. I also added a callback to show rowData state after the edit occurs to show where I think the problem is. Note, license_key is my own trial enterprise license key.

```"""
Example Group Cell Renderer Configuration
https://www.ag-grid.com/archive/{{ag-grid-version}}/react-data-grid/group-cell-renderer/

"""
import dash_ag_grid as dag
from dash import Dash, html
from dash.dependencies import Input, Output, State
from ag_grid_license import license_key

app = Dash(__name__)

data = {
    "Ireland": ["Dublin", "Galway", "Cork"],
    "UK": ["London", "Bristol", "Manchester", "Liverpool"],
    "USA": ["New York", "Boston", "L.A.", "San Fransisco", "Detroit"],
    "MiddleEarth": ["The Shire", "Rohan", "Rivendell", "Mordor"],
    "Midkemia": ["Darkmoor", "Crydee", "Elvandar", "LaMut", "Ylith"],
}

rowData = []
for country, cities in data.items():
    for city in cities:
        rowData.append(
            {
                "country": country,
                "type": "Non Fiction"
                if country in ["Ireland", "UK", "USA"]
                else "Fiction",
                "city": city,
            }
        )

columnDefs = [
    # this column shows just the country group values, but has not group renderer, so there is no expand / collapse functionality
    {
        "headerName": "Country Group - No Renderer",
        "showRowGroup": "country",
        "minWidth": 250,
    },
    # same as before, but we show all group values, again with no cell renderer
    {
        "headerName": "All Groups - No Renderer",
        "showRowGroup": True,
        "minWidth": 240,
    },
    # add in a cell renderer
    {
        "headerName": "Group Renderer A",
        "showRowGroup": True,
        "cellRenderer": "agGroupCellRenderer",
        "minWidth": 220,
    },
    # add in a field
    {
        "headerName": "Group Renderer B",
        "field": "city",
        "showRowGroup": True,
        "cellRenderer": "agGroupCellRenderer",
        "minWidth": 220,
    },
    # add in a cell renderer params
    {
        "headerName": "Group Renderer C",
        "field": "city",
        "minWidth": 240,
        "showRowGroup": True,
        "cellRenderer": "agGroupCellRenderer",
        "cellRendererParams": {
            "suppressCount": True,
            "checkbox": True,
            'innerRenderer': "SimpleCellRenderer",
            "suppressDoubleClickExpand": True,
            "suppressEnterExpand": True,
        },
    },
    {"headerName": "Type", "field": "type", "rowGroup": True},
    {"headerName": "Country", "field": "country", "rowGroup": True},
    {"headerName": "City", "field": "city", "editable": True},
]


@app.callback(
    Output("grid-edit-cell-info", "children"), 
    Input("group-cell-renderer-example", "cellValueChanged"),
    State("group-cell-renderer-example", "rowData"),
    State("group-cell-renderer-example", "virtualRowData"),
    prevent_initial_call=True
)
def update(cell_changed, row_data, virtual_row_data):
    return f"{cell_changed}"


app.layout = html.Div(
    [
        dag.AgGrid(
            id="group-cell-renderer-example",
            columnDefs=columnDefs,
            rowData=rowData,
            columnSize="autoSize",
            defaultColDef={"resizable": True},
            enableEnterpriseModules=True,
            licenseKey = license_key,
            dangerously_allow_code=True,
            dashGridOptions={
                "groupSelectsChildren": True,
                "groupDisplayType": "custom",
                "groupDefaultExpanded": 1,
                "rowSelection": "multiple",
            }
        ),
        html.Div(id="grid-edit-cell-info"),
    ],
    style={"margin": 20},
)

if __name__ == "__main__":
    app.run_server(debug=True)
```

Screenshots to show sequence of events:

1. Cell for City is modified from Ylith to 5555
<img width="1133" alt="image" src="https://github.com/plotly/dash-ag-grid/assets/45929880/23eb234b-a34f-4e0a-944d-4414defe94a6">

2. Blank rows appear at the end, in their own group:
<img width="1130" alt="image" src="https://github.com/plotly/dash-ag-grid/assets/45929880/dce16959-9656-46cd-beec-36761bd05055">

3. Pausing at the post-edit callback reveals blanks are introduced into row_data:
<img width="519" alt="image" src="https://github.com/plotly/dash-ag-grid/assets/45929880/29e8b194-7956-4acb-b9a9-99543478e2ca">

These seem to correspond to the "holes" in the grid where the cells are blank b/c they've been grouped, as if row_data is interpreting what is literally on the screen.

4. Note, this is not related to handling of unbalanced groups - https://www.ag-grid.com/angular-data-grid/grouping-unbalanced-groups/. If you enable groupAllowUnbalanced = true, it simply puts the blanks on separate rows, ungrouped

**Comment by thiasby** [comment]
I am facing the very same issue, trying to edit rows which are children of a parent row aggregate. My example has 12 rows aggregates, and each cell edit leads to the generation of 12 blank rows / None. Likewise in your example @scottmac991 , you have 7 row aggregates and 7 blank rows. 

It looks like the issue is coming from row aggregates indexing, however I was not able to fix it properly. Adding 'getRowId': "params.data.id" in dashGridOptions prevents the creation of blank rows, however it throws the following callback error : 
Cannot read properties of undefined (reading 'series')

**Comment by BSd3v** [comment]
Hello @scottmac991 and @thiasby,

This will take some time to investigate, I'm not exactly sure what is going on between the grid and props that would be causing this issue, since the grid's prop of `rowData` should be an exact copy of what the `grid` says it is.

**Comment by scottmac991** [comment]
Hi @BSd3v, thanks for taking a look. I can report that I also tried @thiasby 's solution of providing a custom row id override using 'getRowId' in grid params and pointing to an appropriate unique id. Interestingly, I found the same outcome whereby the java code throws an error of 'Cannot read properties of undefined( reading 'id'), but only on the first edit event. The second and subsequent attempts to edit the child cells in fact works without exception.
